### ISSUE TITLE
\Doctrine\ORM\Query\Expr\Base::count() doesn't handle dynamic arguments

### ISSUE THREAD

**Issue by zacharylund** [phpstan/phpstan-doctrine#496]
I have been appreciating the recent PHPStan Doctrine query builder improvements. Thank you!

I found an issue with using `count()` for or and and expressions when arguments are added conditionally:

```php
$or = $qb->expr()->orX();

\PHPStan\dumpType($or->count()); // Dumped type: 0

$or->add('a.foo = 1');

\PHPStan\dumpType($or->count()); // Dumped type: 1

if (rand(0, 1)) {
    $or->add('a.bar = 2');
}

\PHPStan\dumpType($or->count()); // Dumped type: 2 (but should be int<1,2>)
```
It turns out that `if ($or->count()) { $qb->andWhere($or); }` can be replaced with `$qb->andWhere($or)` because the empty expression with be reduced, but using `$or->count()` to return `null` instead of executing a query causes this PHPStan error.

**Comment by janedbal** [comment]
Question is if we should try to be that precise in `count` method, because this:

```
$or = new Orx();

$this->maybeFillOrCondition($or),

if ($or->count() > 0) {

}
```

results in 

`Comparison operation ">" between 0 and 0 is always false.`

I'm not sure if this is even solvable, so I'd rather return only `int`.

**Comment by ondrejmirtes** [comment]
Yeah, that would be fine by me too.

**Comment by janedbal** [comment]
This should be solved since [1.3.52](https://github.com/phpstan/phpstan-doctrine/releases/tag/1.3.52)

**Comment by zacharylund** [comment]
Thank you! And thank you @janedbal for your enhancements to this repo.
